### PR TITLE
tbolt/3277 Update /roles endpoint to return correct roles

### LINKED ISSUE #3277: [Feature] Update the roles returned by the roles endpoint
**Description and related issues -**
We need to restrict the role types that are returned by the roles endpoint.

### Acceptance criteria

| Given | When | Then |
| ----- | ---- | ---- |
| a state admin | requesting roles | return ['eAPD State Staff', 'eAPD State Contractor'] |
| a fed admin | requesting roles | return ['eAPD Federal Admin', 'eAPD State Admin'] |

### This task is done when…

This could include:

- [ ] all acceptance criteria are met

### PR DESCRIPTION
**Pull request by tbolt**
resolves #3277 

**Description-**

* Updates the `/roles` endpoint to only return specific roles based on the type of user

**Steps to manually verify this change...**

Use the acceptance criteria in the original ticket: https://app.zenhub.com/workspaces/cms-eapd-5a3823ad3659e94d8ee569be/issues/cmsgov/eapd/3277

### This pull request is ready to review when...

- [x] Automated tests are updated (and all tests are passing)
- [x] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [x] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: api/db/roles.js [hunk 1]
```diff
@@ -4,18 +4,23 @@ const getAllActiveRoles = async (userRole, { db = knex } = {}) => {
   const roles = await db('auth_roles')
     .select('id', 'name')
     .where({ isActive: true });
-  if (userRole === 'eAPD Federal Admin') {
-    return roles.filter(
-      role =>
-        role.name !== 'eAPD System Admin' && role.name !== 'eAPD Federal Admin'
-    );
+  
+  switch (userRole) {
```
Comment (knollfear) [inline api/db/roles.js:8]: I am always wary of doing filtering on results outside of the DB.  The DB is really really good at filtering, we should let it do filtering where we can.  These filters are not super complicated and can definitely be implemented in knex with .whereIn('name', [foo, bar, baz]) 

`
const query =  db('auth_roles')
    .select('id', 'name')
    .where({ isActive: true });

  switch (userRole) {
    case 'eAPD System Admin':
      return query;

`

**Review by knollfear — APPROVED** [review]
This looks good.  If you want to rewrite this to filter in the DB query I will re-review, otherwise I think this is acceptable as is and filtering this in the DB is not a requirement.

File: api/routes/roles/get.js [hunk 2]
```diff
@@ -3,11 +3,17 @@ const { getAllActiveRoles: gr } = require('../../db');
 const { can } = require('../../middleware');
 
 module.exports = (app, { getAllActiveRoles = gr } = {}) => {
+  const accessibleRoles = {
+    'eAPD System Admin': null,
+    'eAPD Federal Admin': ['eAPD Federal Admin', 'eAPD State Admin'],
+    'eAPD State Admin': ['eAPD State Staff', 'eAPD State Contractor']
+  }
   app.get('/roles', can('view-roles'), async (req, res, next) => {
+    console.log('req.user', req.user);
```
Comment (knollfear) [inline api/routes/roles/get.js:12]:
```suggestion

```

File: api/routes/roles/get.endpoint.js [hunk 3]
```diff
@@ -16,11 +16,31 @@ describe('Roles endpoint', () => {
     unauthenticatedTest('get', '/roles');
     unauthorizedTest('get', '/roles');
 
+    it('returns expected number of roles given a system admin', async () => {
+      const authedClient = login('all-permissions');
+      const response = await authedClient.get('/roles');
+      expect(response.status).toEqual(200);
+      expect(response.data.length).toBe(6);
```
Comment (knollfear) [inline api/routes/roles/get.endpoint.js:23]: I think this might be best as a snapshot

Comment (tbolt) [reply api/routes/roles/get.endpoint.js:23]: Yep, this is a good case for snapshots.

**Comment by thetif** [comment]
Works as expected, test by:

|Given|When|Then|Result|
|---|---|---|---|
|a fed admin user|requesting /roles|received ['eAPD Federal Admin', 'eAPD State Admin']|:white_check_mark: PASS|
|a state admin user|requesting /roles|received ['eAPD State Staff', 'eAPD State Contractor']|:white_check_mark: PASS|

**Review by jeromeleecms — APPROVED** [review]
Dev team signed off; I'm good with this.

**Pull request merged**